python/graphstorm/inference/np_infer.py

```diff
@@ -168,3 +171,90 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
             save_node_prediction_results(shuffled_preds, save_prediction_path)
         barrier()
         sys_tracker.check('save predictions')
+
+
+class GSGnnNodePredictionRealtimeInferrer(GSInferrer):
+    """ Inferrer for real-time node prediction tasks on SageMaker endpoints.
+
+    .. versionadded:: 0.5
+        Add `GSGnnNodePredictionRealtimeInferrer` class in v0.5 to support real-time node-level
+        inference on Amazon SageMaker endpoints.
+
+    The real-time inferrer has three major differences from the node prediction offline inferrer:
+
+    1. directly setting and using a DGL graph as input to extract features.
+    2. using DGL dataloader, instead of GS dataloaders that rely on DGL distributed graphs.
```

Comment:
Shall we wrap DGL dataloader with GSInferDataloaders? So we can define the APIs for custom RealtimeInferrer.

Reply:
Yes, create a new `GSgnnRealtimeInferNodeDataLoader` class. Also allow users to use DGL `DataLoader` as the two classes return the same outputs.